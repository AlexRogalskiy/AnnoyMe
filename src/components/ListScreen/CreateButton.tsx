import React, { useRef, useCallback } from "react";
import { FloatingAction } from "react-native-floating-action";
import * as colors from "../../constants/colors";

interface Props {
  onPress: () => void;
}

export default ({ onPress }: Props) => {
  const actionRef = useRef<any>(null);

  const pressAndReset = useCallback(() => {
    onPress();
    setTimeout(actionRef.current.reset, 500);
  }, [onPress, actionRef]);

  return (
    <FloatingAction
      position="right"
      color={colors.green400}
      showBackground={false}
      onPressMain={pressAndReset}
      ref={actionRef}
    />
  );
};
