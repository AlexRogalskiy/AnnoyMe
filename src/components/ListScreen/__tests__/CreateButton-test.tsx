import React from "react";
import { FloatingAction } from "react-native-floating-action";
import renderer, { act } from "react-test-renderer";
import CreateButton from "../CreateButton";

it("CreateButton renders button", () => {
  const tree = renderer
    .create(<CreateButton onPress={() => undefined} />)
    .toJSON();

  expect(tree).toMatchSnapshot("CreateButton-button");
});

it("CreateButton handles clicks", () => {
  jest.useFakeTimers();

  const onPress = jest.fn(() => undefined);

  const component = renderer.create(<CreateButton onPress={onPress} />).root;

  const button = component.findByType(FloatingAction);
  expect(button).toBeTruthy();

  act(() => {
    button.props.onPressMain();
    jest.runAllTimers();
  });

  expect(onPress.mock.calls).toEqual([[]]);

  jest.useRealTimers();
});
