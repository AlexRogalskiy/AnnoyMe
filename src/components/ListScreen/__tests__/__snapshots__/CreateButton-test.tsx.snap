// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateButton renders button: CreateButton-button 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 0,
      "elevation": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 0,
    }
  }
>
  <View
    accessibilityLabel="Floating Action Button"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#9ccc65",
        "borderRadius": 28,
        "bottom": 30,
        "elevation": 5,
        "height": 56,
        "justifyContent": "center",
        "opacity": 1,
        "overflow": "visible",
        "position": "absolute",
        "right": 30,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 3,
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
          Object {
            "scale": 1,
          },
        ],
        "width": 56,
        "zIndex": 2,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 28,
          "height": 56,
          "justifyContent": "center",
          "opacity": 1,
          "width": 56,
          "zIndex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "flex": 1,
            "height": 56,
            "justifyContent": "center",
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
            "width": 56,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "position": "absolute",
                  "width": 2,
                },
                Object {
                  "height": 15,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "height": 2,
                  "position": "absolute",
                },
                Object {
                  "width": 15,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
